src/compilersupport_p.h

```diff
@@ -33,6 +33,14 @@
 #include <stddef.h>
 #include <stdint.h>
 
+#if __STDC_VERSION__ >= 201112L || __cplusplus >= 201103L || __cpp_static_assert >= 200410
+#  define cbor_static_assert(x)         static_assert(x, #x)
+#elif !defined(__cplusplus) && defined(__GNUC__) && (__GNUC__ * 100 + __GNUC_MINOR__ >= 406)
+#  define cbor_static_assert(x)         _Static_assert(x, #x)
+#else
+#  define cbor_static_assert(x)         ((void)sizeof(struct { int m : 2*!!(x) - 1; }))
```

Comment:
I tried that, but neither Clang nor ICC complain about an array of size -1. 

Original code:
```c
#  define cbor_static_assert(x, msg)        (void)(sizeof(char[(!!(msg) * !!(x)) ? 1 : -1]))
```

If you remove that (void) and return cbor_static_assert(0, "Hello") from function, ICC:

```asm
        movl      $-1, %eax
        ret       
```

Clang:
```asm
        movl    $4294967295, %eax       # imm = 0xFFFFFFFF
        popq    %rbp
        retq
```

So I copied the solution from the kernel (BUILD_BUG_ON).